### ISSUE TITLE
LCD Script (chronometer) - Displaying different data than OS or WEBIF

### ISSUE THREAD

**Issue by cybrnook** [pi-hole/pi-hole#674]
##### Expected Behaviour:

Blocking, Queries, Pi-holed values match between webif, fresh OS login, and chronometer script running on LCD screen.
##### Actual Behaviour:

"Queries" value resets every day and remains at a very low value such as 2 or 3 until a reboot. Even though WEBIF or fresh login to the OS reflects the true "Queries" value, in the thousands.
##### Steps to reproduce this issue:

run chronometer script on LCD screen following the pi-hole guide, and wait approximately 24 hours or so.
https://pi-hole.net/faq/install-the-real-time-lcd-monitor-chronometer/
##### Forum Thread Reference:

https://pi-hole.net/forums/topic/lcd-script-chronometer-displaying-different-data-than-os-or-webif/
##

**Comment by PromoFaux** [comment]
Hey, we're just about to push release 2.9.1 which _should_ fix this. I've had reports from one user on reddit of it working, so time will tell!

[This is the Pull request that added the fix](https://github.com/pi-hole/pi-hole/pull/655)

give it about 10 mins and then run `pihole -up` from the command line to update to 2.9.1! (you can check if 2.9.1 is released by running `pihole -v`)

**Comment by cybrnook** [comment]
Gentlemen, and scholars!

Thanks a bunch, will give it a shot tonight when I am home and then of course give feedback tomorrow once we pass the 24 hour mark (though I suspect its the roll of midnight, but only guessing).

**Comment by PromoFaux** [comment]
The fix was found aagges ago by @gh0stShell, but somehow it missed the master branch! Such is git.

Which just goes to show how few people actually use the chronometer on LCD because it's only been in the last week that people have brought it up! 

Question actually, were you using it before 2.9, and 2.9 broke it? Or have you only just started using it? 

And yes, it's when the cron job for `pihole -f` runs at 23:58. Although that is now 00:00 in 2.9.1.

**Comment by cybrnook** [comment]
Hey! No complaints from me, as far as I see I opened the ticket yesterday and you offered the fix the next day :+1: 

I will say that up until now, I have been using my own dns poisoning script that I have been writing for the past few months. That works well, but is localized to the router only. Your solution is much more thick than mine and I am now using both.

With that said, I only recently started playing with the rasp-pi and decided to dedicate one to pi-hole. So, if 2.9 was the current version for, let's say....... , the past 2 weeks, then yes I would assume it was broken in 2.9. I have to admit, I am guilty of not keep an eye too much on release version.

However, I will add that this has been the typical behavior for the past 2 weeks or so. I did run a pihole -up yesterday, and it seemed to bring in 5 new items for update. However after yesterday's update, again this morning the same values.

I will test again this evening with fingers crossed. I think the LCD screen idea is awesome, even if it is a niche market implementation.

**Comment by PromoFaux** [comment]
>However, I will add that this has been the typical behavior for the past 2 weeks or so. I did run a pihole -up yesterday, and it seemed to bring in 5 new items for update.

There would have been no changes yesterday. 2.9.1 is the first release we've pushed since 2.9 (when `pihole -up` was introduced)

We should probably fix the output of that... Basically what `pihole -up` does is synchronise the local repo in `/etc/.pihole` with the repo on github. So any branches that have been added or removed since then will get reflected. The only time any changes actually mean anything is when they're to the `master` branch. I think it just needs to be changed to `git pull master` instead of `git pull`.

I like chronometer, but rarely use it myself! I keep meaning to come up with a version for a 20x4 LCD screen, or even for the lcd on [my keyboard](http://support.logitech.com/en_us/product/gaming-keyboard-g510)!

**Comment by cybrnook** [comment]
Wan Wan Wan (sad trombone), the chronometer script reset itself again this am and is showing queries back at 0 even though there were 50~+- pi-holed entries.

Perhaps it's an issue because we are securing and locking a user session by executing the script during login. Then, in the background the file that the script is reading is being modified. Perhaps a restart of the user session as part of the cron update is in order (if possible)?

The reason I say this, is that even though the LCD screen displays 0, the WEBIF and a FRESH OS login report correctly. Even though they should all be reading from the same place. Somehow the LCD screen session is getting lost in translation.

**Comment by PromoFaux** [comment]
I've not actually tried this myself yet.. but does having chronometer running in an SSH session suffer from this issue?

**Comment by cybrnook** [comment]
Sorry PromoFaux, I will be back into testing this hopefully by this weekend. Late, but not forgotten :-)

**Comment by jacobsalmela** [comment]
I think the PR above will work, but might need some others to test it.

**Comment by cybrnook** [comment]
Sounds great, if needed I can setup an LCD screen this weekend to test it out.

How could I obtain the latest changed to chronometer to test it? Should I just download chromometer.sh directly from github and upload it manually?

**Comment by jacobsalmela** [comment]
You can do it that way, but it's probably easier to try this since the script doesn't really care where it lives:
```
cd ~
curl -o chronometer.sh https://raw.githubusercontent.com/pi-hole/pi-hole/bugfix/chronometer-domain-count/advanced/Scripts/chronometer.sh
chmod 755 chronometer.sh
./chronometer.sh
```
Then press Ctrl+C to stop it at any time.

Remove the script when done:
```
rm ~/chronometer.sh
```

**Comment by cybrnook** [comment]
Thanks jacob and team. I will try and test this out shortly and let the team know.

**Comment by conradstorz** [comment]
I've just tested the chronometer script from above and it still reports double the number of blocked sites as reported on the web console.  Pi-hole Version v2.9.5 Web Interface Version v1.4.4.2

**Comment by PromoFaux** [comment]
This needs to be released to master to work due to some name changes on some global variables

**Comment by PromoFaux** [comment]
You could try changing the names in the one you downloaded to match those in `/etc/pihole/setupVars.conf`

**Comment by jacobsalmela** [comment]
Should be fixed via #891
